contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,654 @@
+  
+  For applications that use block storage natively (like MongoDB) no additional configuration is required as the mount path passed
+  to the application provides the device which MongoDB then uses for the storage path in the configuration file (dbpath). Specific
+  tuning for each application to achieve the highest possibly performance is provided as part of its recommended configurations.
+  
+  Specific use cases around improved usage of storage consumption are included in the use cases listed below as follows:
+  * An admin wishes to expose a block volume to be consumed as a block volume for the user  
+  * An admin wishes to expose a block volume to be consumed as a block volume for an administrative function such 
+    as bootstrapping 
+  * A user wishes to utilize block storage to fully realize the performance of an application tuned to using block devices
+  * A user wishes to read from a block storage device and write to a filesystem (big data analytics processing)
+  Future use cases include dynamically provisioning and intelligent discovery of existing devices, which this proposal sets the 
+  foundation for more fully developing these methods. 
+
+ 
+# Design Overview
+
+  The proposed design is based on the idea of leveraging well defined concepts for storage in Kubernetes. The consumption and 
+  definitions for the block devices will be driven through the PVC and PV definitions. Along with Storage
+  Resource definitions, this will provide the admin with a consistent way of managing all storage. 
+  The API changes proposed in the following section are minimal with the idea of defining a volumeMode to indicate both the definition
+  and consumption of the devices. Since it's possible to create a volume as a block device and then later consume it by provisioning
+  a filesystem on top, the design requires explicit intent for how the volume will be used.
+  The additional benefit of explicitly defining how the volume is to be consumed will provide a means for indicating the method
+  by which the device should be scrubbed when the claim is deleted, as this method will differ from a raw block device compared to a 
+  filesystem. The ownership and responsibility of defining the retention policy shall be up to the plugin method being utilized and is
+  not covered in this proposal.
+  
+  Limiting use of the volumeMode to block can be executed through the use of storage resource quotas and storageClasses defined by the 
+  administrator.
+  
+  To ensure backwards compatibility and a phased transition of this feature, the consensus from the community is to intentionally disable
+  the volumeMode: Block for external provisioners until a suitable implementation for provisioner versioning has been accepted and 
+  implemented in the community. This requirement is better described in the design PR discussion and will be implemented as a separate
```

Comment:
It will be hard for someone reading this doc to reference back to the PR comments/discussions.  We can just say it'll be handled separately.

Reply:
done